src/pool/worker.rs

```diff
@@ -39,27 +39,52 @@ where
     }
 
     pub fn run(mut self) {
-        self.runner.start(&mut self.local);
-        while !self.local.core().is_shutdown() {
-            let task = match self.pop() {
-                Some(t) => t,
-                None => continue,
-            };
-            self.runner.handle(&mut self.local, task.task_cell);
+        if !self.local.runnable() {
+            self.local.sleep();
         }
-        self.runner.end(&mut self.local);
+        self.runner.start(&mut self.local);
+        loop {
+            self.runner.reentrant_start(&mut self.local);
+            while !self.local.core().is_shutdown() {
+                let task = match self.pop() {
+                    Some(t) => t,
+                    None => continue,
+                };
+                self.runner.handle(&mut self.local, task.task_cell);
+                if !self.local.runnable() {
+                    break;
+                }
+            }
+            self.runner.reentrant_end(&mut self.local);
+
+            // If this worker should go to sleep, spawn all futures to remote
+            while let Some(t) = self.local.pop(false) {
+                if self.local.core().is_shutdown() {
```

Comment:
Should be checked before the loop.

Reply:
I think it should be checked in a loop, because this loop is mainly to spawn unprocessed tasks from the local queue to the global queue during scale down. If shutdown occurs during this process, we stop the spawn process and discard it directly.

Otherwise, I think there will be an endless loop, local will spawn tasks to the global, and then continue to steal all tasks from the global.

Reply:
There is a check below this loop, and that check is for “pure shutdown scenarios”

Reply:
> I think there will be an endless loop,

Given that steal is false, how can it be an endless loop?

Reply:
No steal is needed when scale down, but steal is needed when shut down.

Reply:
>> I think there will be an endless loop,

>Given that steal is false, how can it be an endless loop?

I was wrong, there will be no endless loop. It's just that there will be some meaningless spawn_remote operations.